File: src/main/java/com/thebluehats/server/core/modules/WorldSelectionManagerModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
import com.thebluehats.server.game.managers.game.WorldSelectionManager;
import com.thebluehats.server.game.managers.game.regionmanager.RegionManager;

import org.bukkit.plugin.java.JavaPlugin;

public class WorldSelectionManagerModule extends AbstractModule {
    @Provides
    @Singleton
    static WorldSelectionManager provideWorldSelectionManager(JavaPlugin plugin, RegionManager regionManager) {
        return new WorldSelectionManager(plugin, regionManager);
    }

    @Override
    protected void configure() {
    }
}

File: src/main/java/com/thebluehats/server/core/modules/WorldSelectionManagerModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.google.inject.Singleton
import com.thebluehats.server.game.managers.world.WorldSelectionManager
import com.thebluehats.server.game.managers.world.regionmanager.RegionManager
import org.bukkit.plugin.java.JavaPlugin

class WorldSelectionManagerModule : AbstractModule() {
    override fun configure() {}

    companion object {
        @Provides
        @Singleton
        fun provideWorldSelectionManager(plugin: JavaPlugin?, regionManager: RegionManager?): WorldSelectionManager {
            return WorldSelectionManager(plugin, regionManager)
        }
    }
}